rest/api_test.go

```diff
@@ -3671,6 +3672,216 @@ func TestUnsupportedConfig(t *testing.T) {
+	reqBody["_revisions"].(map[string]interface{})["start"] = 2
+	temp, _ := json.Marshal(reqBody)
+	str := string(temp)
+	response := rt.SendRequest("PUT", "/db/doc1?new_edits=false", str)
+	var body db.Body
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docRevId2 := body["rev"].(string)
+	assert.Equal(t, "2-two", docRevId2)
+
+	assertStatus(t, response, http.StatusCreated)
+
+	reqBody = make(map[string]interface{})
+	reqBody["_rev"] = "2-twoa"
+	reqBody["_revisions"] = make(map[string]interface{})
+	reqBody["_revisions"].(map[string]interface{})["ids"] = [2]string{"twoa", docRevDigest}
+	reqBody["_revisions"].(map[string]interface{})["start"] = 2
+	temp, _ = json.Marshal(reqBody)
+	str = string(temp)
+	response = rt.SendRequest("PUT", "/db/doc1?new_edits=false", str)
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docRevId2a := body["rev"].(string)
+	assert.Equal(t, "2-twoa", docRevId2a)
+
+	assertStatus(t, response, http.StatusCreated)
+
+	//Create an attachment
+	attachmentContentType := "content/type"
+	reqHeaders := map[string]string{
+		"Content-Type": attachmentContentType,
+	}
+
+	//Put attachment on doc1 rev 2
+	attachmentBody := base64.StdEncoding.EncodeToString(make([]byte, 20))
+	response = rt.SendRequestWithHeaders("PUT", "/db/doc1/attach1?rev=2-two", attachmentBody, reqHeaders)
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docRevId3 := body["rev"].(string)
+
+	assertStatus(t, response, http.StatusCreated)
+
+	//Put attachment on doc1 conflicting rev 2a
+	attachmentBody = base64.StdEncoding.EncodeToString(make([]byte, 21))
+	response = rt.SendRequestWithHeaders("PUT", "/db/doc1/attach1a?rev=2-twoa", attachmentBody, reqHeaders)
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docRevId3a := body["rev"].(string)
+
+	assertStatus(t, response, http.StatusCreated)
+
+	//Setup
+	reqHeaders = map[string]string{
+		"Accept": "application/json",
+	}
+
+	var bodyRes map[string]interface{}
+
+	//Get and do minor update on rev2
+	response = rt.SendRequestWithHeaders("GET", `/db/doc1?rev=`+docRevId3, "", reqHeaders)
+	json.Unmarshal(response.Body.Bytes(), &bodyRes)
+	delete(bodyRes, "_rev")
+	fmt.Println(bodyRes)
+	temp, _ = json.Marshal(bodyRes)
+	str = string(temp)
+	assertStatus(t, response, http.StatusOK)
+
+	response = rt.SendRequest("PUT", "/db/doc1?rev="+docRevId3, str)
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docRevId4 := body["rev"].(string)
+	assertStatus(t, response, http.StatusCreated)
+
+	//Get and do minor update on rev2a
+	response = rt.SendRequestWithHeaders("GET", `/db/doc1?rev=`+docRevId3a, "", reqHeaders)
+	json.Unmarshal(response.Body.Bytes(), &bodyRes)
+	delete(bodyRes, "_rev")
+	temp, _ = json.Marshal(bodyRes)
+	str = string(temp)
+	assertStatus(t, response, http.StatusOK)
+
+	response = rt.SendRequest("PUT", "/db/doc1?rev="+docRevId3a, str)
+	json.Unmarshal(response.Body.Bytes(), &body)
+	docRevId4a := body["rev"].(string)
+	assertStatus(t, response, http.StatusCreated)
```

Comment:
The test appears to be correctly returning a document with attach1a, as the last thing it does is create a child of docRevId3a which then becomes the winning revision.  I don't see anything wrong with this test as written.